fhd_core/beam_modeling/beam_image.pro

```diff
@@ -55,98 +57,121 @@ ENDIF
+                  (*psf.beam_gaussian_params[pol_i,gi_ref[gi]])[*,fbin],model_npix=psf.pix_horizon*1.3,$
+                  model_res=(2.*obs.kpix*dimension)/psf.pix_horizon*(0.5/obs.kpix))*group_n[gi_use[gi]]
             ENDFOR
             beam_single/=Total(group_n[gi_use])
-            IF Keyword_Set(abs) THEN beam_single=Abs(beam_single)
-            beam_base_uv1=Complexarr(dimension,elements)
-            beam_base_uv1[xl:xh,yl:yh]=beam_single
-;            beam_base_uv1+=Shift(Reverse(reverse(Conj(beam_base_uv1),1),2),1,1)
-            beam_base_single=fft_shift(FFT(fft_shift(beam_base_uv1),/inverse));/2.
-            beam_base+=Real_part(beam_base_single*Conj(beam_base_single));>0
-            n_bin_use+=1.*freq_norm[fi]
-        ENDFOR
-    ENDIF ELSE BEGIN
-        IF N_Elements(n_freq) EQ 0 THEN n_freq=psf.n_freq
-        IF N_Elements(freq_i_use) EQ 0 THEN freq_i_use=findgen(n_freq)
-        nf_use=N_Elements(freq_i_use)
-        freq_bin_use=freq_bin_i[freq_i_use]
-        fbin_use=freq_bin_use[Uniq(freq_bin_use,Sort(freq_bin_use))]
-        nbin=N_Elements(Uniq(freq_bin_use,Sort(freq_bin_use)))
-        FOR bin0=0L,nbin-1 DO BEGIN
-            fbin=fbin_use[bin0]
-            nf_bin=Float(Total(freq_bin_use EQ fbin))
-            beam_single=Complexarr(psf_dim,psf_dim)
+            beam_base+=nf_bin*beam_single*beam_single
+        ENDIF ELSE BEGIN
+            ;
             FOR gi=0,n_groups-1 DO BEGIN
                 beam_single+=Reform(*(*beam_arr[pol_i,fbin,gi_ref[gi]])[rbin,rbin]*group_n[gi_use[gi]],psf_dim,psf_dim)
             ENDFOR
             beam_single/=Total(group_n[gi_use])
             IF Keyword_Set(abs) THEN beam_single=Abs(beam_single)
             beam_base_uv1=Complexarr(dimension,elements)
             beam_base_uv1[xl:xh,yl:yh]=beam_single
-;            beam_base_uv1+=Shift(Reverse(reverse(Conj(beam_base_uv1),1),2),1,1)            
             beam_base_single=fft_shift(FFT(fft_shift(beam_base_uv1),/inverse));/2.
-;            neg_inds=where(real_part(beam_base_single) LT 0,n_neg)
-;            IF n_neg GT 0 THEN beam_base_single[neg_inds]=0.
-            beam_base+=nf_bin*Real_part(beam_base_single*Conj(beam_base_single));>0
-            n_bin_use+=nf_bin*freq_norm[fbin]
             
-        ENDFOR
-    ENDELSE
+            beam_base+=nf_bin*Real_part(beam_base_single*Conj(beam_base_single));>0
+        ENDELSE
+        n_bin_use+=nf_bin*freq_norm[fbin]
+    ENDFOR
+
 ENDIF ELSE BEGIN
-    IF N_Elements(freq_bin_i) EQ 0 THEN BEGIN
-        n_freq_bin=psf.n_freq
+    IF N_Elements(n_freq) EQ 0 THEN n_freq=N_Elements(freq_bin_i)
+    IF N_Elements(freq_i_use) EQ 0 THEN freq_i_use=findgen(n_freq)
+    nf_use=N_Elements(freq_i_use)
+
+    IF keyword_set(beam_gaussian_params) THEN BEGIN
+        ;
+        ; Build the image directly, hence purely real arrays.
+        beam_base_uv=Fltarr(dimension,elements) ;misnomer warning: using _uv to reduce code bulk
```

Comment:
It can....I didn't want to overuse `beam_base`. It's the image for one logic pathway and `beam_base_uv` is the FT of the image in another logic pathway. 

I can make it `beam_base` if you don't think there will be confusion